### PR TITLE
kittycad-kcl-lsp: init at 0.1.61

### PR DESCRIPTION
**Pull request by jljox**
kittycad-kcl-lsp: init at 0.1.61

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/kc/kcl-language-server/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "kcl-language-server";
```
Comment (superherointj) [inline pkgs/by-name/kc/kcl-language-server/package.nix:8]: There is a name clash with [KCL language](https://www.kcl-lang.io/).

I suggest you name it `kittycad-kcl-lsp`.

(This for top-level name of package.)

File: pkgs/by-name/kc/kcl-language-server/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    repo = "kcl-lsp";
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-87S+8uyACWx6URGJGcFdfPbH6UJnR+7685zFSof27Do=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/kc/kcl-language-server/package.nix:22]:
```suggestion
  meta = {
```
Avoid nested with, whenever possible.

Comment (superherointj) [reply pkgs/by-name/kc/kcl-language-server/package.nix:22]: meta.changelog is missing

> meta.changelog = "https://github.com/KittyCAD/kcl-lsp/releases/tag/v${version}";

**Comment by superherointj** [comment]
Commit message should be:

> kittycad-kcl-lsp: init at 0.1.61

And then, include release notes for it at commit message body:

> Release: https://github.com/KittyCAD/kcl-lsp/releases/tag/v0.1.61

There should be no "version" in commit title or PR title.

I suggest you rename PR to: `kittycad-kcl-lsp`

**Comment by superherointj** [comment]
Also squash commits. Ideally it should be a single commit per package.

File: pkgs/by-name/kc/kcl-language-server/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (superherointj) [inline pkgs/by-name/kc/kcl-language-server/package.nix:7]:
```suggestion
}:

rustPlatform.buildRustPackage rec {
```

Would you mind adding a space here?

File: pkgs/by-name/kc/kcl-language-server/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-87S+8uyACWx6URGJGcFdfPbH6UJnR+7685zFSof27Do=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with lib; {
+    description = "The kcl language server";
```
Comment (superherointj) [inline pkgs/by-name/kc/kcl-language-server/package.nix:23]:
```suggestion
    description = "The kcl language server";
```
This description needs to be improved to make clear which KCL LSP is this. KittyCAD, is unrelated to Kubernetes KCL (https://www.kcl-lang.io/).

Comment (Aleksanaa) [reply pkgs/by-name/kc/kcl-language-server/package.nix:23]: And not starting with `A`, `An`, `The`. See https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/by-name/kc/kcl-language-server/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    homepage = "https://github.com/KittyCAD/kcl-lsp";
+    license = licenses.mit;
+    maintainers = [ ];
```
Comment (Aleksanaa) [inline pkgs/by-name/kc/kcl-language-server/package.nix:26]: New packages must have maintainers.

**Comment by superherointj** [comment]
In case you are seeking KCL (Kubernetes) language server, I have it packaged here, and I was waiting for kcl package to be reviewed before upstreaming it.

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 5 above)
Comment (jljox) [reply pkgs/by-name/kc/kcl-language-server/package.nix:26]: Thanks @Aleksanaa for the feedback. I'm new to contributing and I'd like some advice on finding a maintainer for this. 
Could you point me to some documentations for this please?

Comment (superherointj) [reply pkgs/by-name/kc/kcl-language-server/package.nix:26]: Since you are adding the package, you should be the maintainer.

**Review by jljox — COMMENTED** [review]
I think I've addressed all review comments. Thanks everyone for the feedback.

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 1 above)
Comment (jljox) [reply pkgs/by-name/kc/kcl-language-server/package.nix:8]: done

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 2 above)
Comment (jljox) [reply pkgs/by-name/kc/kcl-language-server/package.nix:22]: done

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 3 above)
Comment (jljox) [reply pkgs/by-name/kc/kcl-language-server/package.nix:7]: done

**Comment by superherointj** [comment]
> I think I've addressed all review comments. Thanks everyone for the feedback.

You have not. Are my comments/reviews not showing up for you?

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 3 above)
Comment (jljox) [reply pkgs/by-name/kc/kcl-language-server/package.nix:7]: ok not done yet :)

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/by-name/kc/kcl-language-server/package.nix:8]: Remember to update the directory location/name to `kittycad-kcl-lsp`.

**Comment by superherointj** [comment]
Please note the path is wrong:

> pkgs/by-name/kc/kittycad-kcl-lsp/package.nix

It should be `/ki/`.

File: pkgs/by-name/kc/kittycad-kcl-lsp/package.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "kcl-lsp";
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-51eFOJnc/GqgXtfVx/omR+KuC7x/oKGGR+s0z6nKXBg=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/kc/kittycad-kcl-lsp/package.nix:23]:
```suggestion
  meta = {
```

File: pkgs/by-name/kc/kittycad-kcl-lsp/package.nix [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    description = "KittyCAD Language Server. This is unrelated to https://www.kcl-lang.io/";
+    homepage = "https://github.com/KittyCAD/kcl-lsp";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jljox ];
+    mainProgram = "kittycad-kcl-lsp";
+    meta.changelog = "https://github.com/KittyCAD/kcl-lsp/releases/tag/v${version}";
```
Comment (superherointj) [inline pkgs/by-name/kc/kittycad-kcl-lsp/package.nix:29]:
```suggestion
    changelog = "https://github.com/KittyCAD/kcl-lsp/releases/tag/v${version}";
```
`meta` is repeated.
Add `changelog` after `description`.

File: pkgs/by-name/kc/kcl-language-server/package.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/by-name/kc/kcl-language-server/package.nix]: Not done!!!

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "kcl-lsp";
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-51eFOJnc/GqgXtfVx/omR+KuC7x/oKGGR+s0z6nKXBg=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:23]:
```suggestion
  meta = {
```

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    description = "KittyCAD Language Server. This is unrelated to https://www.kcl-lang.io/";
+    homepage = "https://github.com/KittyCAD/kcl-lsp";
+    license = licenses.mit;
```
Comment (superherointj) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:26]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    maintainers = with maintainers; [ jljox ];
```
Comment (superherointj) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:27]:
```suggestion
    maintainers = with lib.maintainers; [ jljox ];
```

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-51eFOJnc/GqgXtfVx/omR+KuC7x/oKGGR+s0z6nKXBg=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with lib; {
+    description = "KittyCAD Language Server. This is unrelated to https://www.kcl-lang.io/";
```
Comment (superherointj) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:24]:
```suggestion
    description = "KittyCAD KCL language server";
```

**Comment by superherointj** [comment]
PR title should be:
> kittycad-kcl-lsp: init at 0.1.61

**Review by superherointj — APPROVED** [review]
PR title is still wrong... But code-wise looks good.

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}";
+    hash = "sha256-VtrR4v0BJWYdoYFDJpWnmVqDhZMlPGm+g9yjxYfcFxQ=";
+  };
+
+  cargoHash = "sha256-51eFOJnc/GqgXtfVx/omR+KuC7x/oKGGR+s0z6nKXBg=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = {
+    description = "KittyCAD Language Server";
```
Comment (superherointj) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:24]: > description = "KittyCAD KCL language server";

Comment (jljox) [reply pkgs/by-name/ki/kittycad-kcl-lsp/package.nix]: thanks

**Comment by jljox** [comment]
@superherointj  I have changed the PR title to `kittycad-kcl-lsp: init at 0.1.61`.
Let me know if it conforms to the guidelines or standards. 
Thanks

**Comment by superherointj** [comment]
The commit that adds yourself as a maintainer should be a different commit.

**Comment by superherointj** [comment]
> @superherointj I have changed the PR title to `kittycad-kcl-lsp: init at 0.1.61`. Let me know if it conforms to the guidelines or standards. Thanks

I would separate the commit adding myself as maintainer. And would title it (for git commit):

> maintainers: add jljox

**Comment by superherointj** [comment]
The maintainer commit should precede the package commit (or else nix eval fails when doing bicep, for example).

Also I said 'commit' not PR. It should be a single PR with two commits.

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "kittycad-kcl-lsp";
+  version = "0.1.61";
+
+  src = fetchFromGitHub {
+    owner = "KittyCAD";
+    repo = "kcl-lsp";
+    rev = "v${version}";
```
Comment (steeleduncan) [inline pkgs/by-name/ki/kittycad-kcl-lsp/package.nix:15]: `rev = "refs/tags/v${version}"` is safer if there is ever a branch named the same as the tag

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ki/kittycad-kcl-lsp/package.nix (hunk 13 above)
Comment (jljox) [reply pkgs/by-name/ki/kittycad-kcl-lsp/package.nix]: Thanks for the tip. I've made the change.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**